test/common/testlib.py

```diff
@@ -531,23 +531,10 @@ def set_input_text(self, selector: str, val: str, append: bool = False, value_ch
             self.wait_val(selector, val)
 
     def set_file_autocomplete_val(self, group_identifier: str, location: str):
-        file_item_selector_template = "{0} li button:contains({1})"
-
-        path = ''
-        index = 0
-        for path_part in filter(None, location.split('/')):
-            self.click("{0} > div .pf-c-select__toggle-button".format(group_identifier))
-            self._wait_present(group_identifier + " .pf-c-select__menu")
-            path += '/' + path_part
-            file_item_selector = file_item_selector_template.format(group_identifier, path)
-            self.click(file_item_selector)
-            if index != len(list(filter(None, location.split('/')))) - 1 or location[-1] == '/':
-                self.wait_val("{0} > div input[type=text]".format(group_identifier), path + '/')
-            else:
-                self.wait_val("{0} > div input[type=text]".format(group_identifier), path)
-            index += 1
-
-        self.wait_val("{0} > div input[type=text]".format(group_identifier), location)
+        self.set_input_text(f"{group_identifier} .pf-c-select__toggle-typeahead", location)
```

Comment:
Hmm, this actually reduces the feature that it tests. Previously it was really checking that clicking on each part of the part it would give you the new correct directory. Now it just tests the immediate value setting. I am fine with this, but would be nice to introduce another single more advanced test for this component.

Reply:
I thought https://github.com/cockpit-project/cockpit/blob/main/test/verify/check-pages#L562 was sufficient for this. It tests descending into subdirs and detection of files much more thoroughly. TBH I didn't figure out how exactly this was broken to suddenly end up with "/var/tmp/" instead of "/tmp/new.key". If you prefer, I can spend another hour or two on that, though.